aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,621 @@
+             data-val-required="The Email Address field is required." 
+             id="Email" name="Email" value="" /> <br>
+      Password: 
+      <input type="password" data-val="true" 
+             data-val-required="The Password field is required." 
+             id="Password" name="Password" /><br>
+      <button type="submit">Register</button>
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+  </form>
+
+The data annotations applied to the ``Email`` and ``Password`` properties generate metadata on the model. The Input Tag Helper consumes the model metadata and produces `HTML5 <https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5>`__ ``data-val-*`` attributes (see :doc:`/mvc/models/validation`). These attributes describe the validators to attach to the input fields. This provides unobtrusive HTML5 and `jQuery <https://jquery.com/>`__ validation. The unobtrusive attributes have the format ``data-val-rule="Error Message"``, where rule is the name of the validation rule (such as ``data-val-required``, ``data-val-email``, ``data-val-maxlength``, etc.) There are also attributes of the form ``data-val-ruleName-argumentName="argumentValue"``, for example, ``data-val-maxlength-max="1024"``.  If an error message is provided in the attribute, it is displayed as the value for the ``data-val-rule`` attribute. 
+
+Expression names
+^^^^^^^^^^^^^^^^^^^^^
+
+The ``asp-for`` attribute value is a `ModelExpression <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Rendering/ModelExpression/index.html>`__ and the right hand side of a lambda expression. Therefore, ``asp-for="Property1"`` becomes ``m => m.Property1`` in the generated code which is why you don't need to prefix with ``Model``.
+
+Collections
+^^^^^^^^^^^^^^
+
+Sample, a model containing an array of ``Colors``:
+
+.. literalinclude::  forms/sample/final/ViewModels/Person.cs
+  :language: c#
+  :lines: 5-10
+  :dedent: 3
+  :emphasize-lines: 4
+
+The action method:
+
+.. code-block:: c#
+
+    public IActionResult Edit(int id, int colorIndex)
+    {
+        ViewData["Index"] = colorIndex;        
+        return View(GetPerson(id));
+    }
+
+The following Razor shows how you access a specific ``Color`` element:
+
+.. literalinclude::   forms/sample/final/Views/Demo/EditColor.cshtml 
+  :language: HTML
+
+The *Views/Shared/EditorTemplates/String.cshtml* template:
+
+.. literalinclude::   forms/sample/final/Views/Shared/EditorTemplates/String.cshtml 
+  :language: HTML
+  
+Sample using ``List<T>``:
+
+.. literalinclude::  forms/sample/final/ViewModels/ToDoItem.cs
+  :language: c#
+  :lines: 3-7
+  :dedent: 3
+
+The following Razor shows how to iterate over a collection:
+
+.. literalinclude::   forms/sample/final/Views/Demo/Edit.cshtml 
+  :language: HTML
+
+Using the *Views/Shared/EditorTemplates/ToDoItem.cshtml* template reduces evaluating the indexer from three to one.
```

Comment:
Remove this sentence. It comes out of left field since the "three" isn't shown. Further the "This template replaces ..." part of the sample below is sufficient.